.github/workflows/build-and-test.yaml

```diff
@@ -27,7 +27,7 @@ jobs:
 
       - name: Compile ${{ github.repository }}
         run:
-          cargo build && cargo build --bin spdm
+          cargo build && cargo build -p spdm
```

Comment:
Could you please also fix the other `cargo build --bin spdm` that exists in the project? (e.g. in the `justfile` and some `*.md` files)